cowprotocol/cowswap
# Fixes incorrect date formatting

# Summary

Fixes #2025

Date picker ends up not picking up a value, due to value being supplied in incorrect formatting (as the assumption of en-CA format for Intl formatter does not result in the expected result).

Updated the formatter to use toISOString, with a timezone adjustment (MDN seems to be suggesting this as a solution as well).

  # To Test

1. Open the page `limit`
2. Press on expiry dropdown
3. Press on Custom
4. Expect;
- [ ] Date picker starts at 30 minutes from current time.
- [ ] Date picker should allow you to change the value.
- [ ] Change should be saveable via pressing the corresponding button.

# Additional comments
  
Issue seems to be because our date format util doesn't do what we think it does (en-CA does not return a yyyy-mm-dd format string). Fixing that should solve the missing information. Thankfully the util is used only for the picker, and comments show the intent so the fix shouldn't hold a risk of regression because of that.

Unfortunately different browsers handle date time picker differently, so showing the time on firefox as a clickable thing as opposed to chrome is not possible in the current approach, however now it should always have a value and solve this issue (user would need to click on hour / minute and type if they want to edit).
